src/v/cluster/partition.cc

```diff
@@ -574,21 +577,19 @@ partition::cloud_storage_timequery(storage::timequery_config cfg) {
     co_return result;
 }
 
-ss::future<std::optional<storage::timequery_result>>
-partition::local_timequery(storage::timequery_config cfg) {
+ss::future<std::optional<storage::timequery_result>> partition::local_timequery(
+  storage::timequery_config cfg, bool allow_cloud_fallback) {
     vlog(clusterlog.debug, "timequery (raft) {} cfg(k)={}", _raft->ntp(), cfg);
 
     cfg.min_offset = _raft->log()->to_log_offset(cfg.min_offset);
     cfg.max_offset = _raft->log()->to_log_offset(cfg.max_offset);
 
-    auto result = co_await _raft->timequery(cfg);
+    vlog(clusterlog.debug, "timequery (raft) {} cfg(r)={}", _raft->ntp(), cfg);
 
-    const bool may_answer_from_cloud
-      = may_read_from_cloud()
-        && _cloud_storage_partition->bounds_timestamp(cfg.time);
+    auto result = co_await _raft->timequery(cfg);
 
     if (result.has_value()) {
-        if (may_answer_from_cloud) {
+        if (allow_cloud_fallback) {
```

Comment:
nit: naming suggestion, `null_below_log_start` -- at least, it doesn't assume anything about how callers will it and describes the behavior. Then callers can comment why they're setting it